Frameworks/QuartzCore/CALayer.mm

```diff
@@ -2495,96 +2290,54 @@ - (void)_setRootLayer:(BOOL)isRootLayer {
-
-        //  Calculate new center point
-        origin.x = curLayer->priv->bounds.size.width * curLayer->priv->anchorPoint.x;
-        origin.y = curLayer->priv->bounds.size.height * curLayer->priv->anchorPoint.y;
-    }
-}
-
-// TODO: GitHub issue 508 and 509
-// We need a type-safe way to do this with projections.  This is copied verbatim from the projections
-// code and works perfectly for this limited usage, but we don't do any type validation below.
-// all _createRtProxy instance needs to go in a shared DLL in the future
-inline id _createRtProxy(Class cls, IInspectable* iface) {
-    // Oddly, WinRT can hand us back NULL objects from successful function calls. Plumb these through as nil.
-    if (!iface) {
-        return nil;
-    }
-
-    RTObject* ret = [NSAllocateObject(cls, 0, 0) init];
-    [ret setComObj:iface];
-    return [ret autorelease];
-}
-
-inline WXUIElement* _getBackingXamlElementForCALayer(CALayer* layer) {
-    Microsoft::WRL::ComPtr<IInspectable> fromNode(GetCACompositor()->GetXamlContentElement([layer _presentationNode]));
-    return _createRtProxy([WXUIElement class], fromNode.Get());
+    return priv->_sublayerXamlElement;
 }
 
 /**
  @Status Interoperable
 */
 + (CGPoint)convertPoint:(CGPoint)point fromLayer:(CALayer*)fromLayer toLayer:(CALayer*)toLayer {
-    CGPoint ret = { point.x, point.y };
-
-    if (fromLayer && toLayer) {
-        // get the backing xaml UIElement for fromLayer
-        WXUIElement* fromLayerElement = _getBackingXamlElementForCALayer(fromLayer);
+    if (!fromLayer && !toLayer) {
+        if (DEBUG_VERBOSE) {
+            TraceVerbose(TAG, L"convertPoint cannot convert points for two nil layers.");
+        }
 
-        // get the backing xaml UIElement for toLayer
-        WXUIElement* toLayerElement = _getBackingXamlElementForCALayer(toLayer);
+        return point;
+    }
 
-        // set up transform from xaml elment in fromLayer to xaml element in toLayer
-        WUXMGeneralTransform* transform = [fromLayerElement transformToVisual:toLayerElement];
+    if (!fromLayer) {
+        // get the top most layer for the current layer hierarchy
+        fromLayer = _findSuperLayerForLayer(toLayer);
+    }
 
-        // transform the points in fromLayer to point in toLayer
-        WFPoint* pointInFromLayer = [WXPointHelper fromCoordinates:point.x y:point.y];
-        WFPoint* pointInToLayer = [transform transformPoint:pointInFromLayer];
-        ret = { pointInToLayer.x, pointInToLayer.y };
+    if (!toLayer) {
+        // get the top most layer for the current layer hierarchy
+        toLayer = _findSuperLayerForLayer(fromLayer);
     }
 
+    // Get the backing sublayer xaml UIElement for fromLayer
+    // We use the sublayer element to support proper point conversion within a scrolled view.
+    WXUIElement* fromLayerElement = [fromLayer _getSublayerXamlElement];
```

Comment:
>WXUIElement* [](start = 4, length = 12)

Could we make this work with WXFrameworkElement instead? Just wondering if we need to keep it consistent for readability or is there something else that is used in the parent class.